### DISCUSSION TITLE
We are working on v3!

### DISCUSSION THREAD

**Discussion by cpvalente** [cpvalente/ontime#720] [Discussion]
We have had a busy year since we last released v2. Currently, there is enough in the pipeline that we decided to bump the release to v3.
Numbers aside, this means:
We will slow down feature development and take some time back in beta releases to get things right
We will make breaking changes to the UI and APIs, meaning consumers will need to update

# Road to v3
The plan is to make the changes progressively to make the transition quick and smooth. 

We will start distributing beta releases as soon as we have unblocked all the breaking changes in the roadmap. 

## Breaking changes:
INTEGRATIONS: Restructure of the emitted data
UI: New structure for Application Settings
UI: New design for the Editor view
DATA: We propose the deprecation of presenter and subtitle fields in favour of all dynamic user fields

These structural changes allow us to provide some features we want for the app. The big ones are:
- Project file management system
- Over / Under
- Linked events
- Secondary timers
- Creation of a new Director View
- Event reporter
- Dynamic user fields
- New documentation, more links to the code rather than outdated docs
- Improve API and companion integration (tell us how)
- Strengthen import of data from spreadsheets, ie: google sheet or excel

## Editor View
We want the editor view to focus on managing the application and the rundown (data entry and control). The editor view is the users first introduction to Ontime, here they should be able to get enough information to get a project going and learn more as they go.

### Editor specification:
- A place to import, add and edit a rundown
- Easy access to playback controls
- Easy management of the messages to different views
- Provide an overview of automation and runtime state
- It should be friendly. A large part of our user base just wants a timer...

### Current modifications include:
Remove the info panel. We will be moving the logs to a new Application Settings window.
Redesign the rundown and event editor (bottom drawer). We need a solution that offers more room for expansion. Users also frequently request easier access to event information while running the show. This should be possible here and fully featured in the Director's view.
Look into having the rundown shown being able to be shown  as a list and as a table (similar to the cuesheet)

## Director View
We have hinted in the past about the need to have a new view for the Director / Showrunner roles. 
There is yet to be a clear direction of what that should be like and its responsibilities. 

I imagine this would be a lightweight view focused on controlling the rundown, close to the operator view, but with added playback controls.
We would also need to add an extra layer of configuration where the director could configure data to track. ie: show me the `title` and `userfield0` of the current event and the `title`, `presenter` and `userfield1` of the upcoming event

We need a better specification here and hope for users' input, but the main idea:
- Less focus on data entry and more on 
- Provide an overview of events and running timers

## Event Reporter
We want to support users in the theatre and stage performance fields.
The new Event Reporter feature will record the runtime - when events start and finish - and allow this to be downloaded after a show to be used as a template for a show report.

## When to expect the first beta
In honesty, any attempt at providing a timeline for v3 is pure speculation and of little value.
Regardless:
If we can keep the current capacity and do not find any significant road bumps, I believe we are looking at an early April release.
To reiterate, this doesn't mean we will deliver all the above features. It just means that we have made the necessary changes to make them possible.

I hope this provides enough context and a place to start a discussion. 

- If something is missing or if you have any input on the above changes, user feedback is welcome both here and by email [mail@getontime.no](mailto:mail@getontime.no) as usual
- if you would like to collaborate in specification, development or testing, same email [mail@getontime.no](mailto:mail@getontime.no)

**Comment by tcconway** [comment]
A few points I'd like to add here, although they are probably covered above in the mentioned larger topics.

I use Ontime as a Show Caller more than not, so these points might be swayed toward that role.

**Time Stretch** (Part of "Over / Under"?)
I'm always adjusting time during a show. Usually when I need to, I wait for the speaker on stage to look away THEN I shave off a minute of their time here and there. I'd love a feature to adjust the time so that the clock slows down (or speeds up) meaning that a true second is not a true second, but calculated to fit the allotted time. That way the presenter doesn't know we changed time (and thereby get distracted). If they walk off stage and go "those 5 minutes went by QUICK!" We'd just shrug and say they did perfect.

**Director View** (Part of that view?)
Typically when I do a corporate show, I:
1. set up a Google Sheets for OnTime, with each segment and time
2. import that into OnTime
3. Back in Google Sheets, add rows with specific cues within each timed segment. Walkup music, lighting looks, etc. all on sequential lines before the next OnTime time change--These cues all take place within the same time allotted segment, 
4. The trouble is that I can't Import this Google Sheets in again without deleting 50-100 lines first.

So, I guess I'm asking if you are up to a way to integrate that somehow.
<img width="1446" alt="Screenshot 2024-01-20 at 11 17 54 AM" src="https://github.com/cpvalente/ontime/assets/403326/94575625-16a8-439b-8c8e-52496c2bc624">

**Reply by cpvalente** [reply]
Hi @tcconway , great input as usual.

### Time Stretch (Part of "Over / Under"?)

I think this is a good feature for a stage timer and I am not opposed to doing it, but I am afraid of the consequences it might have for other features that we have planned. Namely secondary timers and external timer sources.

Lets get through v3 and revisit
@alex-Arc is particularly resourceful with working through these kinds of things

### Director View (Part of that view?)

Better integration with spreadsheet workflows is definitely top of the list.
As far as I understand from your request we are after:
- ability to import blocks
- ability to skip import of a row

The current proposal of using `timer-type` column to achieve this is our best running. 

I propose that `timer-type` could have other values defined by the user.
On import, in addition to the already defined ones, we could look for a value of `block` - in which case we make a block - or something that the user defines, such as `private`. In this last case, we dont know what it is, so we would skip importing that row data.

How does this sound? Is there something else on import that I overlooked?

Thank you again

**Comment by lygilygi** [comment]
good to see v3 is coming! I have 1 single request. Would it be possible to add simple progress bar at the bottom of the screen? I run it  From docker on PI and wold be great to be able to a) remove  clock from top right corner and b) put a simple progress bar bottom screen. Sometimes less is more ;)

**Reply by cpvalente** [reply]
Thank you @pawellygan

I assume you are talking about the presenter timer in the /timer route?
You should already be able to achieve this with the customisations available in the panel.
Maybe I have misunderstood?

![timer-custom](https://github.com/cpvalente/ontime/assets/34649812/3e1b8a07-8dba-408b-a2a5-cf915c8f9692)

**Reply by lygilygi** [reply]
Yes that is correct but with headless raspberry pi you can't change it from editor panel - that's my point 

Would be awesome to have "customise" panel accessible from editor view

**Reply by lygilygi** [reply]
Or... you can used customised url to get the view you want - that is more than enough

**Reply by cpvalente** [reply]
Glad you sorted it out.
As you mentioned, all the configurations in a page are in the URL.

From the editor view you can use the aliases feature to create a customised URL with your options
https://ontime.gitbook.io/v2/features/url-aliases

Hope this helps

**Comment by cpvalente** [comment]
v3 is now in beta, so we will close this discussion
